### ISSUE TITLE
Allow a selection of text to be parsed in the playground

### ISSUE THREAD

**Issue by NickLarsenNZ** [pest-parser/site#60]
Currently the whole text is parsed, which means when debugging single rules, you have to keep deleting and re-adding sections of the input text.

It would be really nice to be able to select lines to parse, along with the rule to use.

![image](https://github.com/user-attachments/assets/f99744ac-d5ab-4d0b-b36a-0187ab63ebbb)

It would be super cool if it worked the same as Github selections (where you can click a line number, hold SHIFT, then click another line number - and the URL also has a selection fragment like `#L7-L15`).

**Comment by LeoDog896** [comment]
CodeMirror doesn't have any explicit plugins for gutter-based line selection, I believe, so there would have to be a home-baked solution. Optionally, it could be based on the selected text in the editor, but that would be much more finicky. I would put the gutter selection inside the share string, though, instead of its own separate hash.

tl;dr the base feature sounds good :+1:

**Comment by nathanielknight** [comment]
I like the idea, but I agree that the implementation would be tricky. It _might_ be easier to replace the editor with a stack of editors and let you break up your text that way instead?
